### PR TITLE
Typing delay: CSharpEditorFormatingService constructor

### PR DESCRIPTION
**Pull request by rchande**
CSharpEditorFormattingService creates an ImmutableDictionary and multiple ImmutableHashSets in its constructor. This shows up in the editor's typing delay telemetry because all of these sets are created when the C#EFS is statisfied by MEF on the UI thread. To ameliorate this:
* Replace _multiWordsMap with a switch statement
* Remove the unused _autoFormattingTriggerChars set
* Construct the _supportedChars set with one call to CreateRange

### CODE REVIEW

**Comment by rchande** [comment]
Tag @dotnet/roslyn-ide for review

**Comment by rchande** [comment]
Addresses DevDiv 239559

File: src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs [hunk 1]
```diff
@@ -29,23 +29,11 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.Formatting
...
-            _supportedChars = _autoFormattingTriggerChars.Union("{}#nte:)");
-
-            // set up multi words map
-            _multiWordsMap = ImmutableDictionary.CreateRange(new[]
-            {
-                KeyValuePair.Create('n', ImmutableHashSet.Create(SyntaxKind.RegionKeyword, SyntaxKind.EndRegionKeyword)),
-                KeyValuePair.Create('t', ImmutableHashSet.Create(SyntaxKind.SelectKeyword)),
-                KeyValuePair.Create('e', ImmutableHashSet.Create(SyntaxKind.WhereKeyword)),
-            });
+            _supportedChars = ImmutableHashSet.CreateRange<char>(";{}#nte:)");
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs]: Can you add a comment to motivate what the heck this list even is?

File: src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs [hunk 2]
```diff
@@ -271,14 +259,17 @@ private bool IsEndToken(SyntaxToken endToken)
 
         private bool ValidSingleOrMultiCharactersTokenKind(char typedChar, SyntaxKind kind)
         {
-            ImmutableHashSet<SyntaxKind> set;
-            if (!_multiWordsMap.TryGetValue(typedChar, out set))
+            switch (typedChar)
```
Comment (jasonmalinowski) [inline src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs]: Can you comment what this code even does? I have no clue what to do if I was asked to modify this.

**Comment by Pilchie** [comment]
:+1: with a couple of the comments @jasonmalinowski suggests.  I know this, but only because I wrote one incarnation of the C# formatting code many moons ago.

**Comment by CyrusNajmabadi** [comment]
So, have we actually established that this change improves perf?  My guess is htat we may be loading System.Immutable here and that's causing a delay.  But even after the change there's still an immutable collection being created in the constructor.

File: src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs]: I don't see how the immutable hashsets and immutable dictionaries could be a problem, but this immutable hashset won't be a problem.

**Comment by rchande** [comment]
retest windows_eta_open_prtest

**Comment by rchande** [comment]
test vsi

**Comment by mattwar** [comment]
@dotnet-bot test windows_eta_open_prtest please

**Pull request merged**